src/main/java/org/kohsuke/github/GitHub.java

```diff
@@ -447,6 +456,23 @@ public boolean isCredentialValid() throws IOException {
     }
 
     /**
+     * Ensures that the API URL is valid.
+     */
+    public boolean isApiUrlValid() throws IOException {
+        try {
+            GHApiInfo apiInfo = retrieve().to("/", GHApiInfo.class);
+            try {
+                new URL(apiInfo.getRateLimitUrl());
```

Comment:
it is an easy way to verify that the endpoint (API URL) replied a right response. [Here](https://api.github.com/) an example.